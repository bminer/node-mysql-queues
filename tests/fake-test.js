var queue = require('../index');

//Create a fake MySQL client
var db = {
	'query': function(sql, input, cb) {
		if(typeof input == "function")
		{
			cb = input;
			input = undefined;
		}
		console.log("Executing " + sql);
		//Simulate a database delay of 1 second
		setTimeout(function() {
			console.log("-------------" + sql + " CB");
			if(cb != null) cb();
		}, 1000);
	}
};

queue(db); //Enable queuing

db.query("1");
db.query("2", function() {
	db.query("2cb 1");
	db.query("2cb 2", function() {
		db.query("2cb 2cb");
	});
});
db.query("3A");

var q1 = db.createQueue();
q1.query("Q1 1");
q1.query("Q1 2", function() {
	db.query("6", function() {
		q1.query("9", function() {
			q1.query("11");
		});
		db.query("8");
		q1.execute();
		q1.query("10");
		db.query("12");
	});
	q1.query("Q1 2cb 1");
	q1.query("Q1 2cb 2", function() {
		q1.query("Q1 2cb 2cb");
	});
});
q1.query("Q1 3");
db.query("3B");
q1.execute();
db.query("4");

//Should do nothing since queue is empty
q1.execute();
q1.execute();
q1.execute();

var q2 = db.createQueue();
q2.query("Q2 1");
q2.query("Q2 2", function() {
	db.query("7");
	q2.query("Q2 2cb 1");
	q2.query("Q2 2cb 2", function() {
		q2.query("Q2 2cb 2cb");
	});
});
q2.query("Q2 3");
q2.execute();
db.query("5");
