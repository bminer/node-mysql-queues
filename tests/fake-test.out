Executing 1
Executing 2
Executing 3A
Executing 3B
Executing Q1 1
Executing Q1 2
Executing Q1 3
-------------1 CB
-------------2 CB
-------------3A CB
-------------3B CB
-------------Q1 1 CB
-------------Q1 2 CB
-------------Q1 3 CB
Executing Q1 2cb 1
Executing Q1 2cb 2
-------------Q1 2cb 1 CB
-------------Q1 2cb 2 CB
Executing Q1 2cb 2cb
-------------Q1 2cb 2cb CB
Executing 4
Executing Q2 1
Executing Q2 2
Executing Q2 3
-------------4 CB
-------------Q2 1 CB
-------------Q2 2 CB
-------------Q2 3 CB
Executing Q2 2cb 1
Executing Q2 2cb 2
-------------Q2 2cb 1 CB
-------------Q2 2cb 2 CB
Executing Q2 2cb 2cb
-------------Q2 2cb 2cb CB
Executing 5
Executing 2cb 1
Executing 2cb 2
Executing 6
Executing 7
-------------5 CB
-------------2cb 1 CB
-------------2cb 2 CB
Executing 2cb 2cb
-------------6 CB
Executing 8
Executing 9
-------------7 CB
-------------2cb 2cb CB
-------------8 CB
-------------9 CB
Executing 10
Executing 11
-------------10 CB
-------------11 CB
Executing 12
-------------12 CB
